helpful.el

```diff
@@ -1945,6 +1938,28 @@ state of the current symbol."
         (insert "\n\n")
         (insert (helpful--make-manual-button helpful--sym))))
 
+    ;; Update the rest of the buffer in a thread, since finding references and
+    ;; keybindings can take some time.
+    (if (fboundp 'make-thread)
+        (make-thread
+         (lambda ()
+           (helpful--update-rest start-line start-column primitive-p look-for-src buf pos)))
+      (helpful--update-rest start-line start-column primitive-p look-for-src buf pos))
+
+    (when opened
+      (kill-buffer buf))))
```

Comment:
No it is not.  I had trouble figuring what was the purpose of that piece of code.  Can we just move it to the end of the thread function?

Reply:
If you're looking at a symbol in helpful, and it's defined in a file that you don't currently have open, helpful needs to open it. We look at the file for the source code and references. If helpful opened the file, it closes it after getting this data, so we don't leave files open that the user isn't looking at.

So yep, you should be able to just move it to the end of the thread function. If the thread is the last thing to use the buffer, that's the best place to clean up this buffer.

Reply:
Then indeed we should move it.  I've updated the PR.

Reply:
@fmdkdd It used to be that files were handled just by out-of-the-way helper functions. Unfortunately, this led to the same file being opened repeatedly, which is both slow and noisy. So the file-opening was moved into the main update function so that it could be called just once, with the resulting output passed around down the line. The compromise, of course, is that doing it that way is inelegant and messy. See https://github.com/Wilfred/helpful/pull/134 for more details.

Reply:
@nickdrozd Thanks for the context.

I've modified two failing tests to wait a bit before checking the buffer since it is now populated asynchronously.